### PR TITLE
138323285-rm_find_frameworks_iter__add_find_supplier_declarations

### PR DESCRIPTION
**Pull request by benvand**
* remove `find_frameworks_iter` (never used, broken)
* add `find_supplier_declarations`
* version bump
* test

### CODE REVIEW

File: dmapiclient/data.py [hunk 1]
```diff
@@ -140,6 +140,12 @@ def register_framework_interest(self, supplier_id, framework_slug, user):
             user=user,
         )
 
+    def find_supplier_declarations(self, supplier_id):
+        response = self._get(
+            "/suppliers/{}/frameworks".format(supplier_id)
+        )
+        return {'declarations': response['frameworkInterest']}
```
Comment (TheDoubleK) [inline dmapiclient/data.py]: We don't generally have the ApiClient muddy the API response at all - we just pass the response as-is. Looks like the only time we do wrap things I a new key is in `get_supplier_declaration`, which I suspect is where you got the idea to do the same thing here. I don't think that's a practice we want to spread though, and this method would be better just passing the `frameworkInterest` response through unmodified.

Comment (TheDoubleK) [inline dmapiclient/data.py]: That should have been "modify" rather than muddy... on my stupid phone :)

Comment (benvand) [inline dmapiclient/data.py]: So I've changed it to just return `response`
Is your suggestion that returning `response['frameworkInterest']` is a better idea?

Comment (TheDoubleK) [inline dmapiclient/data.py]: No just the passing the whole response is the right thing I think.

**Comment by TheDoubleK** [comment]
:+1:

**Pull request merged**